### DISCUSSION TITLE
Consumption broken down by half hour

### DISCUSSION THREAD

**Discussion by m315426** [BottlecapDave/HomeAssistant-OctopusEnergy#225] [Ideas]
Hi,

Great integration, it has been very helpful so far. I have had a look at the Octopus API myself and see that consumption data can be returned in half hour readings. Is this being worked on to be available soon or is there a guide to develop for the integration?

Thanks

**Comment by BottlecapDave** [comment]
Hello there and thanks for using the integration.

The sensors have a breakdown currently available as an attribute and are already available in half hour increments. Because the data isn't available in real time, I am limited in how it's presented in Home Assistant out of the box. The easiest way was to present it as a day's accumulation.

I'm [currently looking](https://twitter.com/BottlecapDave/status/1644012220794241025?s=20) at ways to import these values historically, but trying to iron out the remaining bugs when importing the data.

**Reply by m315426** [reply]
Thanks, is there a way to build upon this investigation you have started and help to develop the integration?

**Reply by m315426** [reply]
No worries, found the branch and got it working, happy to test and fix any bugs that are outstanding

**Reply by BottlecapDave** [reply]
Sorry for the very late response, but glad you found the branch. I'm still trying to track down the issue reported in my tweet.

**Reply by m315426** [reply]
Is it that the total accumulative consumption is calculated at that time and added to the state? And that the breakdown statistic should be a separate entity? I have the same thing happening on my setup, here is yesterday and today for me 
![SmartSelect_20230502_173206_Home Assistant](https://user-images.githubusercontent.com/131400057/235749482-e0fd4812-1728-4818-9561-09b5512129f1.jpg)
![SmartSelect_20230502_173216_Home Assistant](https://user-images.githubusercontent.com/131400057/235749504-607dd327-3574-4d56-947b-8250106f6108.jpg)

**Reply by BottlecapDave** [reply]
I was trying to keep it within the same entity, but I don't think it's possible in my scenario I'm wanting to supress the storing of the main state change (which I don't think is possible). HA supports external statistics which I was going to optionally support but I think I'm going to have to settle for that implementation only. Going to try it out over the next couple of days.

**Reply by m315426** [reply]
I have removed the self._state, self._last_reset and self._attributes assignments from async_update() as I only wanted to see the breakdowns for the previous day and not the accumulated consumption, however I still get a negative value of the total accumulative consumption for the first hour of today, what do I do to remove that?

![image](https://user-images.githubusercontent.com/131400057/236202252-80966976-83e9-4d26-bfc6-feeabdedc208.png)

**Reply by BottlecapDave** [reply]
Hi and sorry for the late reply. I've had to go a different route for this, and the fix has been released in v7.1.0. There are instructions in [energy dashboard guide](https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/blob/develop/_docs/energy_dashboard.md) on how to use these new stats.

**Reply by m315426** [reply]
Thanks Dave, works great so far!
